src/rgw/driver/rados/rgw_log_backing.cc

```diff
@@ -609,7 +609,6 @@ bs::error_code logback_generations::remove_empty(const DoutPrefixProvider *dpp,
     if (ec) return ec;
     auto tries = 0;
     entries_t new_entries;
-    std::unique_lock l(m);
```

Comment:
you shouldn't be removing this pair of lock/unlock

Reply:
Done

Reply:
@ronen-fr  I don't understand this. Are you talking about lock-unlock on line 649 and 622 respectively?

Reply:
@Suyashd999 I mean that the code between 612 & 622 was under lock in the original code, and
now it is not.

Reply:
> @Suyashd999 I mean that the code between 612 & 622 was under lock in the original code, and now it is not.

We are not locking anywhere between the lines of code 612 - 622. After declaration of the lock we are directly unlocking it. I don't think there is any need of the lock-unlock pair. Please correct me if this isn't the case.